client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -33,10 +33,22 @@ module.exports = React.createClass( {
 		}
 
 		window.scrollTo( 0, 0 );
-		analytics.tracks.recordEvent( 'calypso_checkout_page_view', { saved_cards: this.props.cards.get().length } );
+		if ( this.props.cart.hasLoadedFromServer ) {
+			this.trackPageView( this.props.cart );
+		}
 		upgradesActions.resetTransaction();
 	},
 
+	componentWillUpdate: function( nextProps ) {
```

Comment:
Because we're unconcerned with the state change (`nextState` is the second argument provided in `componentWillUpdate`), `componentWillReceiveProps` might be a more appropriate lifecycle method to use here, and is generally what we use when reacting to a props change.

Also, what do you think about commenting this behavior, and maybe switching to a single condition? It wasn't immediately obvious to me why we need this check (might just be me, though :) ).

```js
if ( ! this.props.cart.hasLoadedFromServer && nextProps.cart.hasLoadedFromServer ) {
  // if the cart hadn't loaded when this mounted, record the page view when it loads
  this.trackPageView();
}
```